core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -140,6 +150,18 @@ impl SchedulerController {
         Ok(())
     }
 
+    fn pre_scheduling_filter(transaction: &SanitizedTransaction, bank: &Bank) -> bool {
+        let mut error_counters = TransactionErrorMetrics::default();
+        bank.check_transactions(
```

Comment:
I vaguely recall calling `check_transactions` transaction by transaction isn't preferred, would increase lock contention to `status_cache` therefore impact other parts of system. This could be dated knowledge, probably worth double checking.

Reply:
[5ac9074](https://github.com/solana-labs/solana/pull/34252/commits/5ac90748887766c7bd0c2a9cc3b1b754c29467b5)

Made a commit here, it complicates the loop a bit but I pulled it out into a closure to try making it a bit simpler. lmk what you think.